src/base/bittorrent/tracker.cpp

```diff
@@ -238,7 +239,7 @@ Http::Response Tracker::processRequest(const Http::Request &request, const Http:
         if (request.method != Http::HEADER_REQUEST_METHOD_GET)
             throw MethodNotAllowedHTTPError();
 
-        if (request.path.toLower().startsWith(ANNOUNCE_REQUEST_PATH))
```

Comment:
Despite what clazy suggested, by looking at the code paths in Qt, I'm under the impression that this would actually be faster than the other one.

Reply:
>looking at the code paths in Qt 

Is generic case-insensitive comparison so inefficient?

Reply:
>Is generic case-insensitive comparison so inefficient?

Good question. It is counter-intuitive that Qt would employ an inefficient code path.
From my intuition (and code inspection), going `Qt::CaseInsensitive` way  would need to normalize/fold both strings while `toLower()` way will only need folding 1 string.

Reply:
>It is counter-intuitive that Qt would employ an inefficient code path.

I don't think this should be so difficult to implement more effectively if my assumption that case normalization of an individual character doesn't require memory allocation is correct.

Reply:
>I don't think this should be so difficult to implement more effectively if my assumption that case normalization of an individual character doesn't require memory allocation is correct.

You are correct about no unnecessary memory allocation for case normalization.
However my second paragraph in the previous post points out that at least our current code should have less work to execute.

Reply:
I still think "case insensitive srartsWith" should be faster, even in our particular case (although I haven't compared their implementations in great detail and haven't measured actual performance).

Reply:
@glassez @Chocobo1 If this change is too controversial, I can removed it for now until there is more consensus, and we can just merge the other commit.

Reply:
>I can removed it

If @Chocobo1 really dislike it. I has no time to compare them deeply.

Reply:
I don't insist on this.